File: original artifact/Ian_Foss_cs-360_project_three_inventory_app/cs360_project_three/app/src/main/java/com/example/cs_360_project_three/InventoryViewHolder.java
package com.example.cs_360_project_three;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class InventoryViewHolder extends RecyclerView.ViewHolder {

    TextView itemName;
    TextView itemId;
    TextView itemQuantity;
    ImageView deleteIcon;


    // Creates viewholders for each item attribute and a delete button
    public InventoryViewHolder(@NonNull View itemView) {
        super(itemView);
        itemId = itemView.findViewById(R.id.item_id);
        itemName = itemView.findViewById(R.id.item_name);
        itemQuantity = itemView.findViewById(R.id.item_quantity);
        deleteIcon = itemView.findViewById(R.id.delete_icon);
    }

    // Binds the data to the viewholders
    public void bind(InventoryItemEntity item, Runnable onDelete) {
        itemId.setText(String.valueOf(item.getId()));
        itemName.setText(item.getName());
        itemQuantity.setText(String.valueOf(item.getQuantity()));
        deleteIcon.setOnClickListener(v -> onDelete.run());
    }
}

File: enhancements/Enhancement One/CS-499_WIP/app/src/main/java/com/example/CS499_Capstone/InventoryViewHolder.kt
package com.example.CS499_Capstone

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class InventoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    var itemName: TextView
    var itemId: TextView
    var itemQuantity: TextView
    var deleteIcon: ImageView


    // Creates viewholders for each item attribute and a delete button
    init {
        itemId = itemView.findViewById<TextView>(R.id.item_id)
        itemName = itemView.findViewById<TextView>(R.id.item_name)
        itemQuantity = itemView.findViewById<TextView>(R.id.item_quantity)
        deleteIcon = itemView.findViewById<ImageView>(R.id.delete_icon)
    }

    // Binds the data to the viewholders
    fun bind(item: InventoryItemEntity, onDelete: Runnable) {
        itemId.setText(item.id.toString())
        itemName.setText(item.name)
        itemQuantity.setText(item.quantity.toString())
        deleteIcon.setOnClickListener(View.OnClickListener { v: View? -> onDelete.run() })
    }
}
